client/components/survey-container/components/question-step-mapping.tsx

```diff
@@ -0,0 +1,58 @@
+import { QuestionType } from '../types';
+import { QuestionSelectionComponentProps } from './question-step';
+import SurveyCheckboxControl from './survey-checkbox-control';
+import SurveyFlowCardControl from './survey-flow-card';
+import SurveyRadioControl from './survey-radio-control';
+
+type questionTypeComponentMapProps = {
+	[ QuestionType.SINGLE_CHOICE ]: ( {
+		onChange,
+		question,
+		value,
+		disabled,
+		onContinue,
+	}: QuestionSelectionComponentProps ) => JSX.Element;
+	[ QuestionType.MULTIPLE_CHOICE ]: ( {
+		onChange,
+		question,
+		value,
+		disabled,
+		onContinue,
+	}: QuestionSelectionComponentProps ) => JSX.Element;
+};
+
+type questionHeaderButtonsMapProps = {
+	[ key: string ]: {
+		shouldHideContinueButton: boolean;
+		shouldHideSkipButton: boolean;
+	};
+};
+
+export const questionTypeComponentMap = {
+	[ QuestionType.SINGLE_CHOICE ]: SurveyRadioControl,
+	[ QuestionType.MULTIPLE_CHOICE ]: SurveyCheckboxControl,
+};
+
+export const flowQuestionMapping: {
+	[ key: string ]: {
+		questionTypeComponentMap: questionTypeComponentMapProps;
+		headerButtonsMapping: questionHeaderButtonsMapProps;
+	};
+} = {
+	'/start/guided/initial-intent': {
+		questionTypeComponentMap: {
+			[ QuestionType.SINGLE_CHOICE ]: SurveyFlowCardControl,
```

Comment:
The initial idea for this mapping was to have an expandable structure so that any type could be added in the future without extra mapping. This mapping for the question component could be replaced by a different question type; for example, it could use a new type like `SINGLE_CHOICE_CARD` or something similar, pointing to the `SurveyFlowCardControl` component. So, it would end up with a mapping like this:
```
export const questionTypeComponentMap = {
	[ QuestionType.SINGLE_CHOICE ]: SurveyRadioControl,
	[ QuestionType.MULTIPLE_CHOICE ]: SurveyCheckboxControl,
	[ QuestionType.SINGLE_CHOICE_CARD ]: SurveyFlowCardControl,
};
```

Reply:
Moving the discussion to this PR: https://github.com/Automattic/wp-calypso/pull/91063